eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -154,38 +160,66 @@ jobs:
     parameters:
       SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
       SubscriptionConfigurations: ${{ parameters.CloudConfig.SubscriptionConfigurations }}
+      SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
+      EnvVars: ${{ parameters.EnvVars }}
 
   - template: /eng/common/TestResources/deploy-test-resources.yml
     parameters:
       ServiceDirectory: ${{ parameters.ServiceDirectory }}
       Location: ${{ coalesce(parameters.Location, parameters.CloudConfig.Location) }}
       SubscriptionConfiguration: $(SubscriptionConfiguration)
+      UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
+      EnvVars: ${{ parameters.EnvVars }}
 
   - template: /eng/common/testproxy/test-proxy-tool.yml
     parameters:
       runProxy: false
 
   - ${{ parameters.PreTestSteps }}
 
-    # For non multi-config generator use the same build configuration to run tests
-    # We don't need to set it to invoke ctest
-    # Visual Studio generator used in CI is a multi-config generator.
-    # As such, it requires the configuration argument for building and invoking ctest
-  - bash: |
-      export AZURE_CLIENT_ID=$(${{parameters.ServiceDirectory}}_CLIENT_ID)
-      export AZURE_TENANT_ID=$(${{parameters.ServiceDirectory}}_TENANT_ID)
-      export AZURE_CLIENT_SECRET=$(${{parameters.ServiceDirectory}}_CLIENT_SECRET)
+  - ${{ if UseFederatedAuth }}:
+    - task: AzurePowerShell@5
+      displayName: ctest
+      # Runs only if test-resources are happily deployed.
+      # unit-tests runs for those configs where samples are not ran.
+      # This enables to run tests and samples at the same time as different matrix configuration.
+      # Then unit-tests runs, samples should not run.
+      condition: and(succeeded(), ne(variables['RunSamples'], '1'))
+      inputs:
+        azureSubscription: ${{ parameters.CloudConfig.ServiceConnection }}
+        azurePowerShellVersion: LatestVersion
+        ScriptType: InlineScript
+        Inline: |
+          $account = (Get-AzContext).Account
+          $env:AZURESUBSCRIPTION_CLIENT_ID = $account.Id
+          $env:AZURESUBSCRIPTION_TENANT_ID = $account.Tenants
```

Comment:
Where is `AZURESUBSCRIPTION_SERVICE_CONNECTION_ID` set?

Reply:
It's set by `AzurePowerShell@5` task's outer script.